lib/internal/loader/ModuleRequest.js

```diff
@@ -36,21 +38,32 @@ loaders.set('esm', async (url) => {
 
 // Strategy for loading a node-style CommonJS module
 loaders.set('cjs', async (url) => {
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading CJSModule ${url}`);
-    const CJSModule = require('module');
-    const pathname = internalURLModule.getPathFromURL(new URL(url));
-    CJSModule._load(pathname);
+  debug(`Loading CJSModule ${url}`);
+  const CJSModule = require('module');
+  const pathname = internalURLModule.getPathFromURL(new URL(url));
+  const exports = CJSModule._load(pathname);
+  const es = Boolean(exports[esModuleInterop] || exports.__esModule);
```

Comment:
I wonder if it might be useful to allow `[esModuleInterop]: false` to cancel out `__esModule` when that isn't wanted?